### PR TITLE
Implement late review suggestions on #14421

### PR DESCRIPTION
**Pull request by eteq**
This is a quick follow-on from #14421 that is basically just what I wanted to suggest but then the PR got merged before I managed to submit my review.  @mhvk is the interested party as the author of #14421 .  I'm also marking it as not needing a changelog because it's really just a minor delta on 14421

Note that, perhaps unsurprisingly, `EarthLocation` does *not* work. But since I wrote the tests I figure we might as well include them as `xfail` since we want those tests to eventually work.

### CODE REVIEW

**Comment by eteq** [comment]
Oh and I think the situation here is that the `EarthLocation.from_geodetic` requires #14423 to work, but `EarthLocation.from_geocentric` is a separate problem.  But @mhvk might want to weigh in if you have more insight.

**Comment by pllim** [comment]
pre-commit.ci autofix

File: astropy/coordinates/angles.py [hunk 1]
```diff
@@ -571,6 +571,7 @@ def _validate_angles(self, angles=None):
         # Ensure ndim>=1 so that comparison is done using the angle dtype.
         # Otherwise, e.g., np.array(np.pi/2, 'f4') > np.pi/2 will yield True.
         # (This feels like a bug -- see https://github.com/numpy/numpy/issues/23247)
+        # TODO: address this again when/if numpy 2.0 exists - see above issue
```
Comment (pllim) [inline astropy/coordinates/angles.py:574]: xref https://github.com/numpy/numpy/issues/23247

**Comment by mhvk** [comment]
My sense would be to move this to 6.0; I really want to make the structured dtype work anyway, and we might then as well built that on top of this. I don't see much advantage to merging this now.

**Comment by pllim** [comment]
If you change the milestone, pls remove the `Affects-dev` label and rethink whether this needs a change log, because it would affect released code. FYI.

**Comment by mhvk** [comment]
I'm 90% of the way with a PR for 6.0 that does structured dtypes (and which includes these commits)...  Removed `affects-dev`...

**Comment by eteq** [comment]
I'm fine with leaving it to 6.0 !

**Comment by mhvk** [comment]
See #15304 for what is actually needed to work with `EarthLocation`. Thankfully, quite minimal changes to `EarthLocation` itself.

**Comment by pllim** [comment]
Superseded by #15304

**Pull request closed without merging**